oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+
+  Removed
+  ~~~~~~~
+
+  * Removed deprecated function `xxx`. See DEPR-XXX for details.
+
+  [2.0.0] - 2020-03-21
+  --------------------
+
+  Added
+  ~~~~~
+
+  * Feature x. See announcement here: https://xxx
+
+Notable Differences
+~~~~~~~~~~~~~~~~~~~
+
+The following are notable differences between the recommendations of this OEP and the `keepachangelog.com`_ recommendations:
+
+* We use GitHub Releases in certain cases, as you'll see below, for developer efficiency purposes.
+* Because we typically release every change for libraries, we don't get any benefit from an "Unreleased" section.
+* We sometimes use automation that uses git commit messages for changelog messages. As noted above, it is acceptable and appropriate to clean these up as needed.
+
+Library Repositories
+--------------------
+
+All Open edX library repositories should contain a CHANGELOG.rst. These libraries should all adhere to `Semantic Versioning`_.
+
+Semantic Versioning
+~~~~~~~~~~~~~~~~~~~
+
+All versioned code should use a MAJOR.MINOR.PATCH format for versioning, as defined by the `Semantic Versioning v2`_ specification. Here is a copy of the most relevant part of the summary:
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+* *MAJOR* version when you make *incompatible API changes*,
+* *MINOR* version when you *add functionality in a backwards compatible manner*, and
+* *PATCH* version when you make *backwards compatible bug fixes*.
+
+Every published version should have a corresponding entry in the changelog, but not necessarily in CHANGELOG.rst.
+
+One of the most important rules when choosing whether to increment the MAJOR, MINOR, or PATCH version, is that backward incompatible changes always result in a new MAJOR version, and never a MINOR or PATCH version.
+
+* You must increment the MAJOR version for backward incompatible changes, even if you think it is unlikely to affect anyone. You can simply note in the changelog entry why it is backward incompatible and why you think it is unlikely to affect anyone.
+* If a backward incompatible change is found in a MINOR or PATCH version after publishing, first update the changelog and communicate immediately.  Next, a new MINOR or PATCH version should immediately be released which reverts this change, and it should be moved to a new MAJOR version.
+* If you decide that reverting will cause more harm for some reason, make sure you clearly communicate this decision.
+
+The reason it is important that only MAJOR versions include backward incompatible change is because when people upgrade dependencies, some will only read changelogs for MAJOR versions because of the potential backward incompatibilities, and may not read notes for MINOR or PATCH upgrades.
+
+Frontend Libraries
+~~~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, frontend libraries should also include a CHANGELOG.rst. However, as you'll see below, this file may simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
```

Comment:
Why is this true particularly for frontend libraries?

Reply:
I made a one word change.  Given that, I'm not sure what you mean by "this".  I'm not seeing where this isn't clear, so I need more information.

Reply:
My question was: why particularly will frontend libraries use a CHANGELOG that simply points to GitHub Releases?  If this is a good solution for them, why isn't it the right choice for other kinds of repos?